pkg/operator/ceph/file/status.go

```diff
@@ -28,20 +28,22 @@ import (
 )
 
 // updateStatus updates a fs CR with the given status
-func (r *ReconcileCephFilesystem) updateStatus(observedGeneration int64, namespacedName types.NamespacedName, status cephv1.ConditionType, info map[string]string) {
+func (r *ReconcileCephFilesystem) updateStatus(observedGeneration int64, namespacedName types.NamespacedName, status cephv1.ConditionType, info map[string]string) error {
 	fs := &cephv1.CephFilesystem{}
 	err := r.client.Get(r.opManagerContext, namespacedName, fs)
```

Comment:
For the error you're seeing, the `observedGeneration` must be different from the generation of this CephFilesystem CR that is retrieved on this line. Try logging both of those values (if set), and also look at the generation of the filesystem CR metadata from kubectl.

Reply:
`ObservedGeneration` is a field in CephFS Status struct / object. When the first call to `updateStatus()` function happens the `Status` field is `nil`, so we don't have any `ObservedGeneration` field (in the Status object). During this first call to `updateStatus()` we hit `The object has been modified` error. Will try to compare the CephFS' before and after objects.